### PR TITLE
Add support to external mappings

### PR DESCRIPTION
**Pull request by gouvermxt**

### CODE REVIEW

**Comment by gouvermxt** [comment]
@Tranquility You will notice that some changes were made by Rubocop formatter that was turned on.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM ruby:2.6.5-buster
+
+# Install ruby gems
+ADD Gemfile /home/mercury/app/Gemfile
+ADD Gemfile.lock /home/mercury/app/Gemfile.lock
+ADD beetle_etl.gemspec /home/mercury/app/beetle_etl.gemspec
+WORKDIR /home/mercury/app
```
Comment (Tranquility) [inline Dockerfile]: No need to use mercury you should be able to just do:

```
COPY Gemfile .
COPY Gemfile.lock .
COPY beetle_etl.gemspec .
```

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: '2.3'
+services:
+  app:
+    build: .
+    volumes:
+      - .:/home/mercury/app
+    links:
+      - db
+    command: tail -f /dev/null
```
Comment (Tranquility) [inline docker-compose.yml]: I don't think you need to have it running all the time. You can just call `rspec` here and execute the command whenever you want to run the tests.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: '2.3'
+services:
+  app:
+    build: .
+    volumes:
+      - .:/home/mercury/app
```
Comment (Tranquility) [inline docker-compose.yml]:
```suggestion
      - .:.
```

File: lib/beetle_etl/steps/table_diff.rb [hunk 4]
```diff
@@ -53,18 +61,25 @@ def transition_update
...
-          ON (stage.external_id = target.external_id)
-        WHERE stage.external_id IS NULL
-        AND target.external_source = '#{external_source}'
-        AND target.deleted_at IS NULL
+          SELECT
+            'DELETE',
+            target.id
+          FROM "#{target_schema}"."#{table_name}" target
+          JOIN "#{target_schema}"."#{mappings_table_name}" AS mapping ON (
+            mapping.external_id = target.external_id
```
Comment (Tranquility) [inline lib/beetle_etl/steps/table_diff.rb]: shouldn't this be `stage` instead of `target`?

**Review by Tranquility — CHANGES_REQUESTED** [review]

File: lib/beetle_etl/steps/step.rb [hunk 5]
```diff
@@ -37,5 +35,12 @@ def stage_table_name(table_name = nil)
       BeetleETL::Naming.stage_table_name(external_source, table_name || @table_name)
     end
 
+    def mappings_table_name(table_name = nil)
+      BeetleETL::Naming.mappings_table_name(table_name || @table_name)
+    end
+
+    def mapped_column_name(table_name = nil)
```
Comment (Tranquility) [inline lib/beetle_etl/steps/step.rb]: The name should communicate better what this is. Maybe something like `mapped_foreign_key_column`.

File: beetle_etl.gemspec [hunk 6]
```diff
@@ -1,15 +1,10 @@
-# coding: utf-8
-lib = File.expand_path('../lib', __FILE__)
-$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
-require 'beetle_etl/version'
-
 Gem::Specification.new do |spec|
   spec.name          = 'beetle_etl'
-  spec.version       = BeetleETL::VERSION
+  spec.version       = '2.0.4'
```
Comment (Tranquility) [inline beetle_etl.gemspec]: I am still wondering why you need this. And it probably should be 3.00.

File: spec/support/database.yml.example [hunk 7]
```diff
@@ -1,9 +0,0 @@
-adapter: 'postgres'
```
Comment (Tranquility) [inline spec/support/database.yml.example:1]: Why did you remove this file?

**Review by Tranquility — CHANGES_REQUESTED** [review]

File: spec/support/database.yml.example (hunk 7 above)
Comment (gouvermxt) [reply spec/support/database.yml.example:1]: I renamed it to database.yml

**Review by Tranquility — APPROVED** [review]
LGTM

**Comment by gouvermxt** [comment]
Duplicate of #2.

**Pull request closed without merging**